test/e2e/fail_forward_e2e_test.go

```diff
@@ -0,0 +1,309 @@
+			Expect(err).To(BeNil())
+
+			err = magicCatalog.UpdateCatalog(context.Background(), fixedProvider)
+			Expect(err).To(BeNil())
+
+			By("waiting for the subscription to have the example-operator.v0.3.0 status.updatedCSV")
+			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasCurrentCSV("example-operator.v0.3.0"))
+			Expect(err).Should(BeNil())
+		})
+
+		It("eventually reports a successful state when using skip ranges", func() {
+			By("patching the catalog with a fixed version")
+			fixedProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fail-forward/skip-range", "example-operator.v0.3.0.yaml"))
+			Expect(err).To(BeNil())
+
+			err = magicCatalog.UpdateCatalog(context.Background(), fixedProvider)
+			Expect(err).To(BeNil())
+
+			By("waiting for the subscription to have the example-operator.v0.3.0 status.updatedCSV")
+			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasCurrentCSV("example-operator.v0.3.0"))
+			Expect(err).Should(BeNil())
+		})
+		It("eventually reports a successful state when using skips", func() {
+			By("patching the catalog with a fixed version")
+			fixedProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fail-forward/skips", "example-operator.v0.3.0.yaml"))
+			Expect(err).To(BeNil())
+
+			err = magicCatalog.UpdateCatalog(context.Background(), fixedProvider)
+			Expect(err).To(BeNil())
+
+			By("waiting for the subscription to have the example-operator.v0.3.0 status.updatedCSV")
+			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasCurrentCSV("example-operator.v0.3.0"))
+			Expect(err).Should(BeNil())
+		})
+		It("eventually reports a failed state when using replaces", func() {
+			By("patching the catalog with a fixed version")
+			fixedProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fail-forward/replaces", "example-operator.v0.3.0.yaml"))
+			Expect(err).To(BeNil())
+
+			err = magicCatalog.UpdateCatalog(context.Background(), fixedProvider)
+			Expect(err).To(BeNil())
+
+			By("waiting for the subscription to maintain the example-operator.v0.2.0 status.updatedCSV")
+			Consistently(func() bool {
+				subscription, err := crclient.OperatorsV1alpha1().Subscriptions(subscription.GetNamespace()).Get(context.Background(), subscription.GetName(), metav1.GetOptions{})
+				if err != nil || subscription == nil {
+					return false
+				}
+				ctx.Ctx().Logf(fmt.Sprintf("%s (%s): %s", subscription.Status.State, subscription.Status.CurrentCSV, subscription.Status.InstallPlanRef))
+				return subscription.Status.CurrentCSV == "example-operator.v0.2.0"
+			}).Should(BeTrue())
```

Comment:
Could it help with errors if instead of returning a boolean, we return the CurrentCSV, then use `.Should(Equal("example-opeartor.v0.2.0`?